### PR TITLE
fix nan while training by conditionally clear input gradient tensor before launch bwd kernel

### PR DESCRIPTION
**Pull request by carlushuang**
This fix [SWDEV-258416](http://ontrack-internal.amd.com/browse/SWDEV-258416), by clear the input gradient buffer before launch bwd kernel

### CODE REVIEW

**Review by zjing14 — APPROVED** [review]
LTGM

**Review by asroy — APPROVED** [review]

File: src/conv/invokers/impl_gemm_dynamic.cpp [hunk 1]
```diff
@@ -241,13 +241,24 @@ InvokerFactory MakeImplGemmDynamicBackwardDataInvokerFactory(const ConvolutionCo
         const int gemm_k_gid = k * y_dot_slice_gid[gemm_id] * x_dot_slice_gid[gemm_id];
         is_gemm_not_empty.emplace_back(gemm_k_gid > 0);
     }
+    bool need_set_zero = false;
+    if(y < stride_h || x < stride_w)
```
Comment (asroy) [inline src/conv/invokers/impl_gemm_dynamic.cpp:245]: Dilation is also a factor. I don't have an exact logic to decide when to set-zero, but it's safe to od

```
if(!(dilation_h == 1 && dilation_w==1 && y >=  stride_h && x >= stride_w))
    need_set_zero = true;
```

**Review by asroy — CHANGES_REQUESTED** [review]

**Comment by asroy** [comment]
@carlushuang  This fix will fix both gfx906 and gfx908?

File: src/conv/invokers/impl_gemm_dynamic.cpp (hunk 1 above)
Comment (carlushuang) [reply src/conv/invokers/impl_gemm_dynamic.cpp:245]: Oh, yes, dilation also have same effect. Let me change it

**Comment by carlushuang** [comment]
@asroy Yes. As long as I use some strategy like bwd v4r1 (split filter into dot, slice, ...), we will always have some pixels of input tensor not get updated

**Review by asroy — APPROVED** [review]

**Review by zjing14 — APPROVED** [review]

**Pull request merged**